airbyte-integrations/connectors/source-orb/integration_tests/catalog.json

```diff
@@ -0,0 +1,144 @@
+{
+  "type": "CATALOG",
+  "catalog": {
+    "streams": [
+      {
+        "name": "customers",
+        "json_schema": {
+          "$schema": "http://json-schema.org/draft-07/schema#",
+          "type": ["null", "object"],
+          "properties": {
+            "id": { "type": "string" },
+            "external_customer_id": { "type": ["string", "null"] },
+            "name": { "type": "string" },
+            "email": { "type": ["string", "null"] },
+            "created_at": { "type": ["null", "string"], "format": "date-time" },
+            "payment_provider": { "type": ["null", "string"] },
+            "payment_provider_id": { "type": ["null", "string"] },
+            "timezone": { "type": ["null", "string"] },
+            "shipping_address": {
+              "type": ["null", "object"],
+              "properties": {
+                "city": { "type": ["null", "string"] },
+                "country": { "type": ["null", "string"] },
+                "line1": { "type": ["null", "string"] },
+                "line2": { "type": ["null", "string"] },
+                "postal_code": { "type": ["null", "string"] },
+                "state": { "type": ["null", "string"] }
+              }
+            },
+            "billing_address": {
+              "type": ["null", "object"],
+              "properties": {
+                "city": { "type": ["null", "string"] },
+                "country": { "type": ["null", "string"] },
+                "line1": { "type": ["null", "string"] },
+                "line2": { "type": ["null", "string"] },
+                "postal_code": { "type": ["null", "string"] },
+                "state": { "type": ["null", "string"] }
+              }
+            }
+          }
+        },
+        "supported_sync_modes": ["full_refresh", "incremental"],
+        "source_defined_cursor": true,
+        "default_cursor_field": ["created_at"],
+        "source_defined_primary_key": [["id"]]
+      },
+      {
+        "name": "subscriptions",
+        "json_schema": {
+          "$schema": "http://json-schema.org/draft-07/schema#",
+          "type": ["null", "object"],
+          "properties": {
+            "id": { "type": "string" },
+            "created_at": { "type": ["null", "string"], "format": "date-time" },
+            "customer": {
+              "type": "object",
+              "properties": { "id": { "type": "string" } }
```

Comment:
Hi @ChristopheDuong that's a great point - if we want to keep just the `customer_id` here, would you recommend that we implement this with a [schema transformer](https://docs.airbyte.com/connector-development/cdk-python/schemas#custom-schema-type-transformation), or is there a different recommended solution? Let me know if there's an instructive example to look at for this sort of transform.

I believe it would be nice to maintain a `customer_id` on the `Subscription` resource rather than removing it because synced Subscriptions in a destination may be more useful with a `customer_id` field.

Reply:
@ChristopheDuong I attempted to look through existing usages of the custom normalizer, and I think it may be possible to do this with by inspecting the field subschema, and flattening the resource if the subschema contains an `id` field. 

Is that the approach you'd recommend? I'm curious if there's a more standardized way to flatten responses as other APIs return nested resources as well.

Reply:
@ChristopheDuong just following up here to see if the use of the custom transform to accomplish this sounds reasonable to you!

Reply:
Sorry, I haven't answered right away as I'm not too familiar with the custom normalizer.

I guess it does sound a reasonable approach to me though.

Reply:
No problem - I'll give that a try!

Reply:
@kgrover the schema transformer only coerces data types to match something declared in the catalog, so it wouldn't achieve the flattening you're after. 

for an example: the google ads connector achieves this via custom code. See [here](https://github.com/airbytehq/airbyte/blob/df5170ad209fb8c1130c2f8e197c501f47352aab/airbyte-integrations/connectors/source-google-ads/source_google_ads/google_ads.py#L97). It's achieved in a bit of a complicated way: fields are declared in the schema JSON with dots in their name e.g: `customer.details.first_name`. The dots denote nesting in the response returned by the API. The connector then reads the field name in the schema, and knows how to recursively denest things from the API.

For now unfortunately your path forward is also via custom transformation. 

I created a relevant issue [here](https://github.com/airbytehq/airbyte/issues/10289) to address this need out of the box in the CDK.